src/main/java/gift/controller/ProductController.java

```diff
@@ -0,0 +1,47 @@
+package gift.controller;
+
+import gift.model.Product;
+import gift.service.ProductService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+@Controller
+public class ProductController {
+
+    @Autowired
+    private ProductService productService;
+
+    @GetMapping("/")
+    public String viewHomePage(Model model) {
+        model.addAttribute("listProducts", productService.getAllProducts());
+        return "index";
+    }
+
+    @GetMapping("/showNewProductForm")
```

Comment:
API를 아래와 같이 하는 경우가 많습니다.

기본적으로 메서드를 통해 행동을 정의내리게 됩니다.
또한 url에 계층형을 두고 의미하는데요

이런 경우엔

```suggestion
    @GetMapping("/products/new")
```
이런식으로 쓰면 이해가 가능할거 같기도 해요~